lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -209,3 +212,76 @@ export function useAPIFetch<T = unknown>(
   const paramStr = recordToQueryString(params ?? {});
   return useFetch(path ? `${path}${paramStr}` : null, fetcher);
 }
+
+export type PaginatedFetchResult<T> = Omit<FetchResult<T>, 'mutate'> & {
+  /** Determines if currently loading the first page */
+  isInitiallyLoading: boolean;
+  /** Loads next page */
+  loadNextPage: () => void;
+};
```

Comment:
I finally decided to make this hook return a `FetchResult` variation, with the next differences:

* It has a `isInitiallyLoading` prop, which is true only when the first page is being loaded. This is equivalent to `isLoading && data === null`, but I found myself doing that in several places, so I decided to include it in the result object itself.
* It has `loadNextPage`, which tries to load the page after current one, only if there's a "next page" to load, which means we already loaded at least the first page, and the server is telling us there's more pages. Otherwise, it's a noop.
* It doesn't have a way to mutate the result (meaning, the `mutate` prop is removed), as I thought that would make things more complicated than they already are. Let's add that capability if/when it's needed.